### ISSUE TITLE
[FEATURE REQUEST] Add the ability to change the icon color

### ISSUE THREAD

**Issue by andrewjswan** [lubeda/EspHoMaTriXv2#174]
# Feature Request

### Describe the solution / feature you'd like

Let's say we have an icon with some image, but we need to display it in different colors (green, yellow, red) depending on the value.
We call the service `icon_set_color(icon_name, r ,g ,b)` which changes all significant points of the icon to the specified color. Those. instead of three (or more) icons, we only need one, and we can display it in different colors.

**Comment by lubeda** [comment]
Hi, why not use the small bitmap command? I don't see the use case. Can you explain i little bit more or give a sample

**Comment by andrewjswan** [comment]
A bitmap is a way out, I agree, but only for static icons, and if we have an animated icon, for example `lameid: 14429`, then we won’t be able to use a bitmap.

**Comment by lubeda** [comment]
ok so you like to manipulate a dual color gif eg. black and red like in 14429 and you will change every red pixel to the specified color?

**Comment by andrewjswan** [comment]
Yes, but not everyone is red, and everyone is not black, i.e. If the color is displayed, change it to the one that is specified.
And so on every call to this service.

**Comment by andrewjswan** [comment]
I'm still looking for a nice solution, so let it hang there for now... Or maybe someone else will find one.

**Comment by lubeda** [comment]
After thoroughly torturing the AI: it's doable — just not comfortable.

```
script:
  - id: show_status
    parameters:
      r: int
      g: int
      b: int
    then:
      - lambda: |-
          // 8x8 shape: 1 = colored pixel, 0 = black
          static const uint8_t mask[64] = {
            0,0,1,1,1,1,0,0,
            0,1,1,1,1,1,1,0,
            1,1,1,1,1,1,1,1,
            1,1,1,1,1,1,1,1,
            1,1,1,1,1,1,1,1,
            1,1,1,1,1,1,1,1,
            0,1,1,1,1,1,1,0,
            0,0,1,1,1,1,0,0,
          };
          uint16_t c = ((r & 0xF8) << 8) | ((g & 0xFC) << 3) | (b >> 3);  // RGB565
          std::string s = "[";
          for (int i = 0; i < 64; i++) {
            if (i) s += ",";
            s += std::to_string(mask[i] ? c : 0);
          }
          s += "]";  
          id(rgb8x32)->bitmap_small(s, "OK", 5, 10, true, 255, 255, 255);
```

**Comment by andrewjswan** [comment]
By the way, can we implement something like that? For example, a function like `change_icon_color(icon_name, r, g, b)`?

**Comment by andrewjswan** [comment]
I'm still looking for a nice solution, so let it hang there for now...
